include/mata/parser/inter-aut.hh

```diff
@@ -123,15 +123,16 @@ public:
  * E.g., a formula q1 & s1 will be transformed to a tree with & as a root node
  * and q1 and s2 being children nodes of the root.
  */
-struct FormulaGraph {
+class FormulaGraph {
+public:
     FormulaNode node{};
     std::vector<FormulaGraph> children{};
 
     FormulaGraph() = default;
-    FormulaGraph(const FormulaNode& n) : node(n), children() {}
-    FormulaGraph(FormulaNode&& n) : node(std::move(n)), children() {}
+    explicit FormulaGraph(const FormulaNode& n) : node(n), children() { children.reserve(2); }
```

Comment:
just wandering, can you put the two alredy to the the line 129 with the declaration?

Reply:
No, vector has no such constructor, unfortunately. It can be done with a lambda or a named function inside the parentheses in the initialization list. Something like
```cpp
FormulaGraph() : children([](){ std::vector<FormulaNode> children; children.reserve(100); return children;}()) {}
```